Update Dependencies: Wagmi, rainbowkit, and their dependencies

ghimiresdp opened the pull request:
- feat: added alchemy provider instead of public
- feat(package): updated dependencies
- refactor(ui): wagmi migrations in components
- feat(UI): updated Rainbowkit, Wagmi, and its implementations and removed SSR issues
- fix: removed unused imports
- fix(faucet): migrate faucet methods to the latest version of wagmi
- fix: update eslint and prettier versions
- fix: update wagmi changes to Answersheet
- fix: update Chakra UI versions

HokutoMorita commented on src/pages/_app.tsx:
```diff
@@ -1,63 +1,39 @@
+
+const config = getDefaultConfig({
+  appName: 'cit-web3-class',
+  projectId: '1b5b1d5c351c838062b62002e37bee3c',
```
Is the projectId value fixed?
Also, what do you need it for?

ghimiresdp replied:
@HokutoMorita 

I am so sorry for the late reply,
the `Rainbow Kit` library needs project ID from `WalletConnect` for integration. The project ID is fixed for a DAPP so we can keep as it is in the source code.

HokutoMorita replied:
I see
Thank you!

HokutoMorita commented:
@ghimiresdp 
Thank you !
I have confirmed that the Polygon Mumbai Error and Next.js render error have been resolved.

HokutoMorita commented on src/components/wallet/ContractDetail.tsx:
```diff
@@ -11,15 +14,15 @@ import {
 
 export const ContractDetail = () => {
   const { t } = useTranslation('common');
   const { address, connector, isConnected } = useAccount();
```
useAccount() is declared on lines 23 and 25.

23 lines
```
const { address, connector, isConnected } = useAccount();
```

25lines
```
const { chain } = useAccount();
```

HokutoMorita replied:
Also, the `connector` variable declared on line 23 is not used.

HokutoMorita commented:
An error has occurred in cit-coin-ui/src/hooks/useTokenBalance.ts.

First:
```
'useNetwork' が宣言されていますが、その値が読み取られることはありません。ts(6133)
モジュール '"wagmi"' にエクスポートされたメンバー 'useNetwork' がありません。ts(2305
```

![スクリーンショット 2024-03-12 14 21 55](https://github.com/henkaku-center/cit-coin-ui/assets/38268537/6f15e2eb-f498-409f-9ad0-2757abdfa2a5)

ghimiresdp commented:
@HokutoMorita 
I see, I will fix it and push new commits for the fixes

HokutoMorita commented:
Thank you !
I haven't been able to review any other files yet.
So I might make some other comments as well.

HokutoMorita commented on src/components/NFT.tsx:
```diff
@@ -54,14 +62,12 @@ const AssetCard = (props: { asset: Asset }) => {
   const { t } = useTranslation('common');
   const [loading, setLoading] = useState(false);
-  const { address, connector, isConnected } = useAccount();
+  const { address, connector, isConnected, chain } = useAccount();
```
Also, the `connector` and `isConnected` variables declared on line 23 are not used.

HokutoMorita commented:
@ghimiresdp 
An error has occurred in `cit-coin-ui/src/components/NFT.tsx`

I'm getting the error "0 arguments expected, 1 specified."
```
0 個の引数が必要ですが、1 個指定されました。
```
![スクリーンショット 2024-03-12 14 52 16](https://github.com/henkaku-center/cit-coin-ui/assets/38268537/99c6fd19-9326-4cb4-860f-1392a20f11c6)
![スクリーンショット 2024-03-12 14 52 07](https://github.com/henkaku-center/cit-coin-ui/assets/38268537/c0b20326-e48f-4587-864e-7c49bc1e1478)

TypeScript type definition may be wrong. ?...

HokutoMorita commented:
@ghimiresdp 
An error has occurred in `cit-coin-ui/src/hooks/useContractConfig.ts`
```
'useNetwork' が宣言されていますが、その値が読み取られることはありません。
```
![スクリーンショット 2024-03-12 14 58 21](https://github.com/henkaku-center/cit-coin-ui/assets/38268537/77c2667b-d967-44dd-b297-3ba022bc9fb3)
`useNetwork` may not be used

HokutoMorita commented:
@ghimiresdp 
An error has occurred in `cit-coin-ui/src/components/Answersheet.tsx`

77 lines
I'm getting the error "0 arguments expected, 1 specified."
```
0 個の引数が必要ですが、1 個指定されました。
```
![スクリーンショット 2024-03-12 15 11 22](https://github.com/henkaku-center/cit-coin-ui/assets/38268537/9de8e849-dfd4-46d5-a18b-b83611116343)

180 lines
I'm getting the error "0 arguments expected, 1 specified."
```
0 個の引数が必要ですが、1 個指定されました。
```
![スクリーンショット 2024-03-12 15 12 09](https://github.com/henkaku-center/cit-coin-ui/assets/38268537/7bb1d214-c37b-4b85-9377-8009e9be84f9)

HokutoMorita commented:
@ghimiresdp 
An error has occurred in `cit-coin-ui/src/pages/faucet.tsx`

49 lines
I'm getting the error "0 arguments expected, 1 specified."
```
0 個の引数が必要ですが、1 個指定されました。
```
![スクリーンショット 2024-03-12 15 15 00](https://github.com/henkaku-center/cit-coin-ui/assets/38268537/1f70320b-2d31-40cd-9170-fd26ed520649)


57 lines
I'm getting the error "The type of parameter 'd' is implicitly 'any'."
```
パラメーター 'd' の型は暗黙的に 'any' になります。
```
![スクリーンショット 2024-03-12 15 16 37](https://github.com/henkaku-center/cit-coin-ui/assets/38268537/950521a7-2102-4122-9fb6-aed9ef7242ca)

HokutoMorita commented:
@ghimiresdp 
There is something additional I found out about this comment. !
https://github.com/henkaku-center/cit-coin-ui/pull/9#issuecomment-1990534346

The `watch` argument for wagmi@2.5.7's useBalance hook may be remove.

v1.x wagmi
https://1.x.wagmi.sh/react/hooks/useBalance
v1.x has a `watch` argument.

v2.x wagmi
https://wagmi.sh/react/api/hooks/useBalance
v2.x does not have a `watch` argument.

ghimiresdp commented:
@HokutoMorita 
I have fixed some issues you previously mentioned. Most of the problems were fixed once I deleted node modules and re ran `yarn` command. can you try it once?

HokutoMorita commented:
> @HokutoMorita I have fixed some issues you previously mentioned. Most of the problems were fixed once I deleted node modules and re ran `yarn` command. can you try it once?

@ghimiresdp 
Please show me your VScode screenshot !

For Example
`cit-coin-ui/src/components/NFT.tsx`
![スクリーンショット 2024-03-15 14 31 23](https://github.com/henkaku-center/cit-coin-ui/assets/38268537/1e96ccf8-a5f0-44e7-ba18-7b8305adc706)


> Most of the problems were fixed once I deleted node modules and re ran yarn command. can you try it once?

I've already tried this method.
But the error remains.

ghimiresdp commented:
@HokutoMorita 
There seems to be no problem in my case

I think you might need to change typescript settings in your VS Code
`settings.json`
```diff
{
-     "typescript.enablePromptUseWorkspaceTsdk": true,
+     // "typescript.enablePromptUseWorkspaceTsdk": true,
}
```

![image](https://github.com/henkaku-center/cit-coin-ui/assets/13810600/f8239b0c-8be7-4521-836d-ebf66c5bef60)
![image](https://github.com/henkaku-center/cit-coin-ui/assets/13810600/deb63bff-bf2b-44ce-a678-282022312a14)

HokutoMorita reviewed (approved):
LGTM！！！

Thank you !!
